src/gtk/toga_gtk/widgets/detailedlist.py

```diff
@@ -1,92 +1,185 @@
-from ..libs import Gtk
+from ..libs import Gtk, GLib, Gdk, Gio
 from .base import Widget
-from .internal.sourcetreemodel import SourceTreeModel
-from .internal.detailedlistrenderers import IconTextRenderer
+from .internal.rows import TextIconRow
+from .internal.buttons import RefreshButton, ScrollButton
 
 
+#Verify if right clicking a row currently works with touch screens, if not, use Gtk.GestureLongPress
 class DetailedList(Widget):
     """
-    gtk DetailedList implementation.
-    This is based on a Gtk.TreeView because the contents is still basic.
-    Should the need arise for more complex contents (buttons on each row, etc.),
-    we should use a Gtk.ListBox (see gtk3-demo).
+    Gtk DetailedList implementation.
+    Gtk.ListBox inside a Gtk.ScrolledWindow.
     """
 
     def create(self):
-        self.renderer = IconTextRenderer.from_style(self.interface)
+        self._on_refresh_handler = None
+        self._on_select_handler = None
+        self._on_delete_handler = None
+        
+        # Not the same as selected row. _active_row is the one with its buttons exposed.
+        self._active_row = None
```

Comment:
Generally, we try to avoid having a Python-side proxy of "state", unless it's completely unavoidable. Is there no way to programatically extract the active row from the ListBox?

Reply:
This is not the selected row (clicked and highlighted in blue), it's the row with the delete button showing. ListBox has no knowledge of this.
The purpose of this variable is hiding the delete button on one row when the user clicks somewhere else. The initial approach I used was a custom signal sent  to all rows `emit on_list_clciked(item: Gtk.ListBoxRow)` but this is not very elegant. If there were 150 rows, all of them would receive a signal, compare the argument to `self` and only one would do something.